skyportal/utils/offset.py

```diff
@@ -516,45 +538,49 @@ def get_nearby_offset_stars(
             queries_issued=queries_issued,
             allowed_queries=allowed_queries,
             use_ztfref=use_ztfref,
+            required_ztfref_source_distance=required_ztfref_source_distance,
         )
 
     # default to Keck starlist
-    sep = ' '  # 'fromunit'
-    commentstr = "#"
-    giveoffsets = True
-    maxname_size = 16
-    # truncate the source_name if we need to
-    if len(source_name) > 10:
-        basename = source_name[0:3] + ".." + source_name[-6:]
-    else:
-        basename = source_name
 
     if starlist_type == 'Keck':
-        pass
+        sep = ' '  # 'fromunit'
+        commentstr = "#"
+        giveoffsets = True
+        maxname_size = 15
+        first_line = None
     elif starlist_type == 'P200':
         sep = ':'  # 'fromunit'
         commentstr = "!"
         giveoffsets = False
         maxname_size = 20
-
-        # truncate the source_name if we need to
-        if len(source_name) > 15:
-            basename = source_name[0:3] + ".." + source_name[-11:]
-        else:
-            basename = source_name
-
+        first_line = None
+    elif starlist_type == 'Shane':
+        sep = ' '  # 'fromunit'
+        commentstr = "#"
+        giveoffsets = True
+        maxname_size = 15
+        # see https://mthamilton.ucolick.org/techdocs/telescopes/Shane/coords/
+        first_line = "!Data {name %16} ra_h ra_m ra_s dec_d dec_m dec_s equinox keyval {comment *}"
     else:
         log("Warning: Do not recognize this starlist format. Using Keck.")
 
-    basename = basename.strip().replace(" ", "")
+    basename = source_name.strip().replace(" ", "")
+    if len(basename) > maxname_size:
+        basename = basename[3:]
+
+    abrev_basename = source_name.strip().replace(" ", "")
+    if len(abrev_basename) > maxname_size - 3:
+        abrev_basename = basename[3:maxname_size]
+
     space = " "
     star_list_format = (
         f"{basename:{space}<{maxname_size}} "
         + f"{center.to_string('hmsdms', sep=sep, decimal=False, precision=2, alwayssign=True)[1:]}"
-        + f" 2000.0 {commentstr}"
+        + f" 2000.0 {commentstr} source_name={source_name}"
```

Comment:
For posterity (and scripting) be sure to add a `source_name` in the comments that has the full, unabbreviated source name from the DB.